upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -762,6 +770,9 @@ func (c *ApplyClusterCmd) Run(ctx context.Context) error {
 
 		// Avoid making changes on a dry-run
 		shouldPrecreateDNS = false
+		if cluster.Spec.GetCloudProvider() == kops.CloudProviderScaleway {
+			shouldPrecreateDNS = true
+		}
```

Comment:
During dry-run should not pre-create DNS records.

Reply:
Sorry I meant to write this line in the Terraform part and not the DryRun part.
This may be temporary as I am considering creating a model and a task for the DNS record resource. But I have not had any conflict if I pre-create DNS records even for Terraform clusters.

Reply:
Could you remove it from here? I don't want to forget about it and have different behaviour from other cloud providers.

Reply:
The model and task for DNS record are ready. I removed the exception for the pre-creation of DNS records with Terraform from `apply_cluster.go`.
Instead the records get pre-created in the kubernetes.tf file with the placeholder IP and with `ignore_changes` on the `data` field because its value will be changed outside of Terraform (similar to the way the etcd volumes are handled)